test/e2e/nodeportlocal_test.go

```diff
@@ -330,9 +363,70 @@ func NPLTestPodAddMultiPort(t *testing.T) {
 
 	checkNPLRulesForPod(t, testData, r, nplAnnotations, antreaPod, testPodIP, true)
 	validatePortsInAnnotation(t, r, nplAnnotations, defaultStartPort, defaultEndPort, targetPorts)
+	validateProtocolsInAnnotation(t, r, nplAnnotations, targetProtocols)
 	checkTrafficForNPL(testData, r, nplAnnotations, clientName)
 
 	testData.deletePod(testNamespace, testPodName)
+	testData.deleteService("agnhost1")
+	testData.deleteService("agnhost2")
+	checkNPLRulesForPod(t, testData, r, nplAnnotations, antreaPod, testPodIP, false)
+}
+
+// NPLTestPodAddMultiProtocol tests NodePortLocal functionalities for a Pod with single port multiple protocols.
+func NPLTestPodAddMultiProtocol(t *testing.T) {
+	r := require.New(t)
+
+	node := nodeName(0)
+	testPodName := randName("test-pod-")
+
+	annotation := make(map[string]string)
+	annotation[k8s.NPLEnabledAnnotationKey] = "true"
+	selector := make(map[string]string)
+	selector["app"] = "agnhost"
+	ipFamily := corev1.IPv4Protocol
+	testData.createServiceWithAnnotations("agnhost1", 80, 8080, defaultProtocolTCP, selector, false, corev1.ServiceTypeClusterIP, &ipFamily, annotation)
+	testData.createServiceWithAnnotations("agnhost2", 80, 8080, svcProtocolUDP, selector, false, corev1.ServiceTypeClusterIP, &ipFamily, annotation)
+	targetPorts := sets.NewInt(8080)
+	targetProtocols := sets.NewString(protocolToString(defaultProtocolTCP), protocolToString(svcProtocolUDP))
+
+	// Creating a Pod using agnhost image to support multiple protocols, instead of nginx.
+	cmd := []string{"/bin/bash", "-c"}
+	args := []string{
+		fmt.Sprintf("/agnhost serve-hostname --udp --http=false --port %d & /agnhost serve-hostname --tcp --http=false --port %d", 8080, 8080),
+	}
+	image := "k8s.gcr.io/e2e-test-images/agnhost:2.29"
+	env := corev1.EnvVar{Name: fmt.Sprintf("SERVE_SCTP_PORT_%d", 8080), Value: "foo"}
```

Comment:
why is there SCTP here?

Reply:
This allowed be to serve both TCP and UDP on the same port using the command
`/agnhost serve-hostname --udp --http=false --port 8080 & /agnhost serve-hostname --tcp --http=false --port 8080`

The other options are `SERVE_{PORT,TLS_PORT,SCTP_PORT}_[NNNN]`
where other options are used to serve only TCP connections. There was also an example I referred to in the existing codebase. 

https://github.com/antrea-io/antrea/blob/7f448b099292a199c36281b32d7dfd32d42f159a/test/e2e/k8s_util.go#L213

Reply:
I understand the `/agnhost serve-hostname --udp --http=false --port 8080 & /agnhost serve-hostname --tcp --http=false --port 8080` command, but I still don't understand this environment variable. Isn't it only consumed by `/agnhost porter` which you don't use here?